modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceTemplateApi.java

```diff
@@ -210,19 +211,49 @@ public List<ServiceTemplateDetailVo> listServiceTemplates(
             @RequestParam(name = "serviceVersion", required = false) String serviceVersion,
             @Parameter(name = "serviceHostingType", description = "who hosts ths cloud resources")
             @RequestParam(name = "serviceHostingType", required = false)
-                    ServiceHostingType serviceHostingType) {
+            ServiceHostingType serviceHostingType) {
         List<ServiceTemplateEntity> serviceEntities =
                 serviceTemplateManage.listServiceTemplates(
-                        categoryName, cspName, serviceName, serviceVersion, serviceHostingType);
-        List<ServiceTemplateDetailVo> serviceTemplateDetailVos =
-                serviceEntities.stream().map(this::convertToServiceTemplateDetailVo).sorted(
-                                Comparator.comparingInt(o -> {
-                                    assert o != null;
-                                    return o.getCsp().ordinal();
-                                }))
-                        .toList();
+                        categoryName, cspName, serviceName, serviceVersion, serviceHostingType,
+                        true);
         log.info(serviceEntities.size() + " service templates found.");
-        return serviceTemplateDetailVos;
+        return getServiceTemplateDetails(serviceEntities);
+    }
+
+
+    /**
+     * List service templates with query params.
+     *
+     * @param categoryName   name of category.
+     * @param cspName        name of cloud service provider.
+     * @param serviceName    name of service template.
+     * @param serviceVersion version of service template.
+     * @return response
+     */
+    @Tag(name = "Service Vendor",
```

Comment:
we need a different tag here -> "Cloud Service Provider"
If possible, we can move this specific method to a new controller also. It would be more cleaner.

Reply:
Updated as suggestion.